### ISSUE TITLE
Add Service Endpoint to retrieve Tuple Shares based on Tags

### ISSUE THREAD

**Issue by kindlich** [carbynestack/amphora#34]
Note: This issue comes from a discussion on carbynestack/ephemeral#17

Currently we have
`intra-vcp/secret-shares` to download a secret's shares based on their secret id.

For allowing to execute ephemeral games using tag-filters, we need an endpoint that

- [ ] Is accessible on the intra-vcp Endpoint `/intra-vcp/**` so that it later on is not authenticated
- [ ] Accepts a Tag-Filter (Query?) Parameter to select all Secrets that have all the provided tags (AND-combined), similar to the currently existing `/secret-shares` endpoint that retrieves MetaData
- [ ] Accepts Sort Property/Direction (Query?) Parameters.
    - This ensures that the output of secrets is the same order on all amphoras across the virtual cloud. 
    - Also, some algorithms may depend on the order (e.g. if it's about time series with moving averages etc.)
    - If not provided by the user, it should be sort by `creation-date` in an `ascending` order
- [ ] Returns the Secret Shares as either
    - A single OutputDeliveryObject containing all Secret Shares concatenated
    - OR a new Transfer Object that is a listing of OutputDeliveryObjects with one per Secret and optionally additional metadata (like the ids of the secrets, etc.)

NOT Part of this issue:

- [ ] Making sure that all endpoints return the same set of secrets (only the order is guaranteed by the endpoint). If in parallel a request deleted/re-tagged secrets so that some of the VCPs include a given secret while others don't, this is currently the user's responsibility.  
    Part of resolving this issue may be adding a Notice to the README, though.

Open questions to discuss:

- [ ] Should this endpoint be HTTP or WebSocket (since it may transmit a lot of data, is HTTP enough?)
- [ ] Should this endpoint offer Paging or just return everything at once?

**Comment by strieflin** [comment]
While this functionality would surely add convenience, it also comes with a lot of complexity. Consensus on a common set and order of input objects is at the heart of this functionality and cannot be decoupled. Until this feature is implemented, filtering of objects can be done client-side.
